src/main/java/teammates/ui/controller/InstructorFeedbackAbstractAction.java

```diff
@@ -0,0 +1,161 @@
+        newSession.setFeedbackSessionName(SanitizationHelper.sanitizeTitle(
+                getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_NAME)));
+        newSession.setFeedbackSessionType(FeedbackSessionType.STANDARD);
+        setTime(newSession);
+        setTimeZoneGracePeriod(newSession);
+        setResultsVisibleFromTime(newSession);
+
+        // handle session visible after results visible to avoid having a
+        // results visible date when session is private (session not visible)
+        setSessionVisibleFromTime(newSession);
+        setEmail(newSession);
+        return newSession;
+    }
+
+    private void setTimeZoneGracePeriod(FeedbackSessionAttributes newSession) {
+        String paramTimeZone = getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_TIMEZONE);
+        if (paramTimeZone != null) {
+            try {
+                newSession.setTimeZone(Double.parseDouble(paramTimeZone));
+            } catch (NumberFormatException nfe) {
+                log.warning("Failed to parse time zone parameter: " + paramTimeZone);
+            }
+        }
+
+        String paramGracePeriod = getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_GRACEPERIOD);
+        try {
+            newSession.setGracePeriod(Integer.parseInt(paramGracePeriod));
+        } catch (NumberFormatException nfe) {
+            log.warning("Failed to parse graced period parameter: " + paramGracePeriod);
+        }
+
+    }
+
+    private void setEmail(FeedbackSessionAttributes newSession) {
+        String[] sendReminderEmailsArray =
+                getRequestParamValues(Const.ParamsNames.FEEDBACK_SESSION_SENDREMINDEREMAIL);
+        List<String> sendReminderEmailsList =
+                sendReminderEmailsArray == null ? new ArrayList<String>()
+                                                : Arrays.asList(sendReminderEmailsArray);
+        newSession.setClosingEmailEnabled(sendReminderEmailsList.contains(EmailType.FEEDBACK_CLOSING.toString()));
+        newSession.setPublishedEmailEnabled(sendReminderEmailsList.contains(EmailType.FEEDBACK_PUBLISHED.toString()));
+        this.setUniqueAttributesForSession(newSession, sendReminderEmailsList);
+    }
+
+    private void setTime(FeedbackSessionAttributes newSession) {
+        newSession.setStartTime(TimeHelper.combineDateTime(
+                getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_STARTDATE),
+                getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_STARTTIME)));
+        newSession.setEndTime(TimeHelper.combineDateTime(
+                getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_ENDDATE),
+                getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_ENDTIME)));
+    }
+
+    private void setResultsVisibleFromTime(FeedbackSessionAttributes newSession) {
+        String type = getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_RESULTSVISIBLEBUTTON);
+        switch (type) {
```

Comment:
Since type is only used once, I recommend inlining the method call, e.g. 
`switch (getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_RESULTSVISIBLEBUTTON))`. Same goes for the other similar method.